services/web/server/tests/unit/with_dbs/test_resource_manager.py

```diff
@@ -541,3 +541,91 @@ async def test_services_remain_after_closing_one_out_of_two_tabs(
     mocked_director_api["stop_service"].assert_has_calls(
         [call(client.server.app, service["service_uuid"])]
     )
+
+
+@pytest.mark.parametrize(
+    "user_role", [(UserRole.USER), (UserRole.TESTER)]
+)
+async def test_websocket_disconnected_none_guests_keep_their_project_and_username(
```

Comment:
instead of creating 2 tests you can add yet another variable such as "expect_call".
since only the 2 last asserts are changing I would recommend doing so to prevent duplication.

Reply:
done, as suggested